external-import/cofense/src/main.py

```diff
@@ -0,0 +1,315 @@
+                object_marking_refs=[stix2.TLP_RED],
+                allow_custom=True,
+            )
+
+            stix_objs.append(incident_relation)
+
+        message = f"Processed ThreatID: {mrti.threat_id}"
+        helper.log_info(message)
+        bundle = stix2.Bundle(*stix_objs, allow_custom=True)
+        self._send_bundle(bundle, work_id=work_id)
+        # self._send_bundle(bundle)
+
+    def _send_bundle(self, bundle: stix2.Bundle, work_id=None):
+        serialized_bundle = bundle.serialize()
+        helper.send_stix2_bundle(
+            serialized_bundle, work_id=work_id, update=update_existing_data
+        )
+        timestamp = int(time.time())
+        helper.set_state({"last_run": timestamp})
+        message = f"Last_run stored, next run in: {round(interval_sec / 60, 2)} minutes"
+        helper.api.work.to_processed(work_id, message)
+
+    # def _send_bundle(self, bundle: stix2.Bundle):
+    #     serialized_bundle = bundle.serialize()
+    #     print(serialized_bundle)
+
+
+def get_position() -> tuple:
+    try:
+        with open("cf_intel.pos", "r") as f:
+            position = f.read().strip()
+            if position:
+                return position, None
+    except Exception as e:
+        helper.log_warning("No position found, defaulting to 30 days ago", e)
+
+    init_date = int((datetime.utcnow() - timedelta(days=30)).timestamp())
+    return None, init_date
+
+
+if __name__ == "__main__":
+    try:
+        while True:
+            try:
+                timestamp = int(time.time())
+                current_state = helper.get_state()
+                if current_state is not None and "last_run" in current_state:
+                    last_run = current_state["last_run"]
+                    helper.log_info(
+                        "Connector last run: "
+                        + datetime.utcfromtimestamp(last_run).strftime(
+                            "%Y-%m-%d %H:%M:%S"
+                        )
+                    )
+                else:
+                    last_run = None
+                    helper.log_info("Connector has never run")
+
+                if last_run is None or (
+                    (timestamp - last_run) > (int(interval_sec) - 1)
+                ):
+
+                    position, init_date = get_position()
+                    connector = CFIntelSync(
+                        CF_USER=user_token,
+                        CF_PASS=user_pass,
+                        INTEGRATION=CofenseIntel,
+                        JITTER=False,
+                        USE_LOCK=False,
+                        INIT_DATE=init_date,
+                        POSITION=position,
+                    )
+                    connector.run()
+
+                else:
+                    new_interval = interval_sec - (timestamp - last_run)
+                    helper.log_info(
+                        "Connector will not run, next run in: "
+                        + str(round(interval_sec / 60, 2))
```

Comment:
With `interval_sec` the displayed interval will always be 60 min (defined by default), I think the idea is to use `new_interval` instead to display the remaining time correctly.
`str(round(interval_sec / 60, 2))` => `str(round(new_interval / 60, 2))`